src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/StackTraceEmissionPolicy.cs

```diff
@@ -28,9 +28,16 @@ public override MethodStackTraceVisibilityFlags GetMethodVisibility(MethodDesc m
     /// </summary>
     public class EcmaMethodStackTraceEmissionPolicy : StackTraceEmissionPolicy
     {
+        private readonly MethodStackTraceVisibilityFlags _flags;
+
+        public EcmaMethodStackTraceEmissionPolicy(bool includeLineNumbers)
+        {
+            _flags = includeLineNumbers ? MethodStackTraceVisibilityFlags.HasLineNumbers : 0;
+        }
+
         public override MethodStackTraceVisibilityFlags GetMethodVisibility(MethodDesc method)
         {
-            MethodStackTraceVisibilityFlags result = 0;
+            MethodStackTraceVisibilityFlags result = _flags;
```

Comment:
Is there any meaning to flags with `HasLineNumbers` without `HasMetadata`? Should this mask out `HasLineNumbers` when `HasMetadata` is 0?